src/ui/Facet/Facet.ts

```diff
@@ -275,8 +277,16 @@ export class Facet extends Component {
      * option, or `occurrences` if no sort criteria is specified.
      */
     sortCriteria: ComponentOptions.buildStringOption({
-      postProcessing: (value, options: IFacetOptions) =>
-        value || (options.availableSorts.length > 0 ? options.availableSorts[0] : 'occurrences'),
+      postProcessing: (value, options: IFacetOptions) => {
+        const logger = new Logger(Facet);
+        const valueIfUndefined = value || (options.availableSorts.length > 0 ? options.availableSorts[0] : 'occurrences');
+        if (valueIfUndefined && QueryUtils.isValidSortCriteria(valueIfUndefined)) {
+          return valueIfUndefined;
+        } else {
+          logger.warn(`The Facet component doesn't accept ${valueIfUndefined} as the value for the sortCriteria option.`);
+          return options.availableSorts.length > 0 ? options.availableSorts[0] : 'occurrences';
```

Comment:
There's still the problem that someone could configure the availableSorts option with invalid value, no ? Making it so this line here could potentially return an invalid sort.

Reply:
That part was already done, I didn't change anything
But it'll be ok, the `value` property which specifies the possible values the list option elements can take is set.
so an invalid value won't be able to be added to the availableSorts